purchase_security/models/purchase_order.py

```diff
@@ -10,9 +13,41 @@ class PurchaseOrder(models.Model):
     is_user_id_editable = fields.Boolean(
         compute="_compute_is_user_id_editable",
     )
+    team_id = fields.Many2one(
+        "purchase.team",
+        string="Team",
+        index=True,
+        auto_join=True,
+        compute="_compute_team_id",
+        store=True,
+        readonly=False,
+    )
 
     def _compute_is_user_id_editable(self):
         is_user_id_editable = self.env.user.has_group(
             "purchase.group_purchase_manager"
         ) or not self.env.user.has_group("purchase_security.group_purchase_own_orders")
         self.write({"is_user_id_editable": is_user_id_editable})
+
+    @api.depends("user_id")
+    def _compute_team_id(self):
+        """When a user is assigned, the first team which the user belongs to is
+        assigned, and if no one, the first purchase team.
+        """
+        first_team = self.env["purchase.team"].search([], limit=1)
+        for record in self:
+            record.team_id = record.user_id.purchase_team_ids[:1] or first_team
+
+    def onchange_partner_id(self):
+        res = super().onchange_partner_id()
+        if self.partner_id:
+            partner = self.partner_id or self.partner_id.commercial_partner_id
```

Comment:
I think it was intended this way?
```suggestion
            partner = self.partner_id.commercial_partner_id
```